tests/unit/subscriptions/test-class-wc-payments-invoice-service.php

```diff
@@ -324,10 +176,52 @@ public function test_maybe_record_first_invoice_payment() {
 		$this->invoice_service->maybe_record_first_invoice_payment( $mock_order->get_id(), 'failed', 'completed' );
 	}
 
+	/**
+	 * Tests for WC_Payments_Invoice_Service::validate_invoice_items()
+	 */
+	public function test_validate_invoice_items() {
+		$mock_order        = WC_Helper_Order::create_order();
+		$mock_subscription = new WC_Subscription();
+		$mock_subscription->set_parent( $mock_order );
+
+		foreach ( $mock_order->get_items( 'line_item', 'fee', 'shipping' ) as $item ) {
+			$item->update_meta_data( self::SUBSCRIPTION_ITEM_ID_META_KEY, 'si_test123' );
+		}
+
+		$mock_items = [
+			[
+				'subscription_item' => 'si_test123',
+				'amount'            => 1000,
+				'quantity'          => 1,
+				'tax_rates'         => [],
+			],
+		];
+
+		$mock_discounts = [];
+
+		$this->mock_product_service
```

Comment:
See this comment: https://github.com/Automattic/woocommerce-payments/pull/2899/files#r712754395